### ISSUE TITLE
Using Qt5AutoUpdater w/ CMake & Qt5.6

### ISSUE THREAD

**Issue by jordanbaucke** [Skycoder42/QtAutoUpdater#7]
I have attempted to add Qt5AutoUpdater to a project with CMake.

Initially, we attempted to just use source files and compile them ourselves. This appears to have some issues on MSVC, so I would prefer to just reference the compiled binaries.

I'm a relative novice with CMake, so any advice would be appreciated:

I have downloaded: [2.1.0](https://github.com/Skycoder42/QtAutoUpdater/releases/tag/2.1.0) (which appears to have `CMake` files included, however, it is targeting Qt 5.8), and then the earlier [2.0.0](https://github.com/Skycoder42/QtAutoUpdater/releases/tag/2.0.0) that do not have any CMake files.

Here is my attempt at a creating a CMakified version: 
https://github.com/jordanbaucke/qtautoupdater_cmake

I removed: 
```
include_directories("")
link_directories("")
```
As I wasn't sure if I needed to "install" the precompiled binaries first - or how to do that without seeing a `CMakeLists.txt` or `Find_Package` module for QtAutoUpdater.

Thanks.

**Comment by Skycoder42** [comment]
I'm very sorry, but I won't be of any help with cmake, because I did not create those files. I never even used cmake. I use the Qt module project structure, and it automatically generates those files as part of the deployment step. And since I don't know cmake, I cannot verify those files.

The reason they appear with 2.1.0 is, that starting there I use the Qt build system. You need to copy the whole module (binaries, headers, etc.) into the Qt installation directory to make it work (This is what the repository does automatically).

If you need them for Qt 5.7, you can try to compile QtAutoUpdater 2.1.0 with Qt 5.7 yourself (using qmake, see [Point 4](https://github.com/Skycoder42/QtAutoUpdater#downloadinstallation)). However, I'm not sure it compiles with 5.7, and I only use the msvc 2015 compilers, I can't say anything for older versions.

If you run the whole build process, it will automatically copy all created binaries, headers, etc. to the correct locations. Once thats done, you should be able to use it as any other Qt Module (core, gui, etc.), even from cmake.

Keep me updated, and if you need help with the compilation of the library itself, feel free to post them here. I may be able to post workarounds, if it's not build with older versions.

**Comment by jordanbaucke** [comment]
@Skycoder42 ,

Thank you for the quick reply. I was able to successfully link the pre-compiled version of the libraries into my app with CMake (using `target_link_libraries`), I updated my [repo](https://github.com/jordanbaucke/qtautoupdater_cmake).

Obviously, it's a stop-gap, and libraries _should be built_ from source and _than link_ the library (but I had to get this done).

If anyone with more CMake knowledge comes along and helps create a `find_package` / installer we could upstream it here for other Qt/CMake people.

Thanks again,

Jordan
